x/storage/keeper/bucket_rate_limit.go

```diff
@@ -0,0 +1,310 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	paymenttypes "github.com/bnb-chain/greenfield/x/payment/types"
+	"github.com/bnb-chain/greenfield/x/storage/types"
+)
+
+func (k Keeper) SetBucketFlowRateLimit(ctx sdk.Context, operator sdk.AccAddress, bucketOwner sdk.AccAddress, paymentAccount sdk.AccAddress, bucketName string, rateLimit sdkmath.Int) error {
+	// check the operator is the same as the payment account owner
+	if !k.paymentKeeper.IsPaymentAccountOwner(ctx, paymentAccount, operator) {
+		return paymenttypes.ErrNotPaymentAccountOwner
+	}
+
+	// get the bucket
+	bucket, found := k.GetBucketInfo(ctx, bucketName)
+
+	// check the bucket owner is the same as the bucket owner of the bucket
+	if found && bucket.Owner != bucketOwner.String() {
+		return fmt.Errorf("the bucket owner is not the same as the bucket owner of the bucket")
+	}
+
+	// if the bucket does not use the payment account, just set the flow rate limit
+	if !found || (found && bucket.PaymentAddress != paymentAccount.String()) {
+		// set the flow rate limit to the store
+		k.setBucketFlowRateLimit(ctx, paymentAccount, bucketName, &types.BucketFlowRateLimit{
+			FlowRateLimit: rateLimit,
+			BucketOwner:   bucketOwner.String(),
+		})
+		return nil
+	}
+
+	// delete the flow rate limit for different bucket owner, since there may be a wrong rate limit or
+	// the previous rate limit set for the bucket which owned by the different bucket owner, for example,
+	// the previous bucket owner delete the bucket and the current bucket owner create a new bucket with the same name
+	k.deleteBucketFlowRateLimitForDifferentBucketOwner(ctx, paymentAccount, bucketOwner, bucketName)
```

Comment:
I strongly suggest that you use the `owner` + `bucketname` + `payment account` as the key of the ratelimit, so that you dont have to delete an already exist `FlowRateLimit`.
Because in some edge case you delete the previous ratelimit, what if there is a bucket still use it.